src/Streamly/Internal/Data/Stream/StreamD/Nesting.hs

```diff
@@ -476,6 +487,827 @@ mergeBy
     => (a -> a -> Ordering) -> Stream m a -> Stream m a -> Stream m a
 mergeBy cmp = mergeByM (\a b -> return $ cmp a b)
 
+{-# INLINE mergeByInner #-}
+mergeByInner
+    :: (MonadIO m,  Eq a, Eq b)
+    => (a -> b -> Ordering) -> Stream m a -> Stream m b -> Stream m (a, b)
+mergeByInner cmp = mergeInnerByM (\a b -> return $ cmp a b)
+
+{-# INLINE_NORMAL mergeInnerByM #-}
+mergeInnerByM
+    :: (MonadIO m,  Eq a, Eq b)
+    => (a -> b -> m Ordering) -> Stream m a -> Stream m b -> Stream m (a, b)
+mergeInnerByM cmp (Stream stepa ta) (Stream stepb tb) = do
```

Comment:
Use the same name as the name of the corresponding join combinator in Top module.